doc/proxy-metrics.md

```diff
@@ -133,3 +135,92 @@ request_total{
 
 [prom-format]: https://prometheus.io/docs/instrumenting/exposition_formats/#format-version-0.0.4
 [pod-template-hash]: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#pod-template-hash-label
+
+# Transport-Level Metrics
+
+The following metrics are collected at the level of the underlying transport
+layer.
+
+### `connection_opens`
+
+A counter of the total number of transport connections which have been opened.
+
+### `connection_closes`
+
+A counter of the total number of transport connections which have been closed.
+
+### `connections`
+
+A gauge of the number of transport connections currently open.
+
+### `sent_bytes`
+
+A counter of the total number of sent bytes.
+
+### `received_bytes`
+
+A counter of the total number of recieved bytes.
+
+### `connection_duration_ms`
+
+A histogram of the duration of the lifetime of a connection, in milliseconds.
+
+### `connection_sent_bytes`
+
+A histogram of the number of bytes sent over the lifetime of a connection.
+This is updated when a connection closes.
+
+### `connection_recieved_bytes`
+
+A histogram of the number of bytes recieved over the lifetime of a connection.
+This is incremented when a connection closes.
+
+## Labels
+
+Each of these metrics has the following labels:
+
+* `direction`: `inbound` if the connection was to/from the proxy's Inbound router,
+               `outbound` if the connection was to/from the proxy's Outbound router.
+* `kind`: `client` if the connection was initiated by the proxy to an upstream server,
+          `server` if the connection was initiated by an upstream client to the proxy.
+* `remote_address`: the IP address and port of the remote end of the connection.
```

Comment:
So, although I did mention using the `dst_` labels for HTTP requests when they're available later in this document, we can only reliably associate them with the proxy's client connections. 

When we see a transport open event that corresponds to someone opening a new connection to the proxy, that event fires before we've detected the protocol or parsed the request. So if that's an outbound request,  that's _way_ before we've gotten as far as looking up the destination for that request. 

This means that we can't easily label, say, a gauge of transports open to the proxy by the destination they're being forwarded to. The only meaningful label I can see for client connections is where they came from, although this could just be inbound vs outbound rather than tracking the inbound connections for individual IPs...